source/io/FileIO.cpp

```diff
@@ -383,6 +391,22 @@ VIREO_FUNCTION_SIGNATURE2(Println, StaticType, void)
 }
 #endif
 //------------------------------------------------------------
+// An instruction to enable the Write to System Log VI which uses an SLI call to SystemLogging_WriteMessageUTF8
+// ignored(0), ignored(1), message(2), severity(3), error cluster(4)
+#if kVireoOS_emscripten
+VIREO_FUNCTION_SIGNATURE5(SystemLogging_WriteMessageUTF8, StringRef, StringRef, StringRef, Int32, ErrorCluster)
```

Comment:
It is better to use VIREO_FUNCTION_SIGNATUREV instead and use Parameter block structure that inherits from InstructionCore where each parameter can be named. See., for example, StringFormat

Reply:
So far `VIREO_FUNCTION_SIGNATUREV` has only been used for functions with a variable number of arguments. Are we making some policy change to use it for all function signatures?

Reply:
Talked to @sanmut and he provided an example not using variable arguments with the StringToByteArray function: https://github.com/ni/VireoSDK/blob/0c290475a41043f52c40b7270e333125ac396ce6/source/core/NumericString.cpp#L2135

Reply:
Much cleaner, thanks @sanmut!

I also created https://github.com/ni/VireoSDK/issues/575 as a spot we should temporarily add information about creating new Vireo instructions. If you can think of anything else that is tribal knowledge making a quick note would be great. :)

Reply:
Sure, thanks :+1: